components/admin/reports/GenerateReportModal.tsx

```diff
@@ -47,26 +47,51 @@ export default function GenerateReportModal<T>(props: GenerateReportProps<T>) {
   const [endDate, setEndDate] = useState('');
   const [step, setStep] = useState(GenerateReportStep.SelectColumns);
 
-  // Checkbox states
-  const [selectAll, setSelectAll] = useState(false);
-  const [userID, setUserID] = useState(false);
-  const [applicantName, setApplicantName] = useState(false);
-  const [applicantDoB, setApplicantDoB] = useState(false);
-  const [APPNumber, setAPPNumber] = useState(false);
-  const [applicationDate, setApplicationDate] = useState(false);
-  const [paymentMethod, setPaymentMethod] = useState(false);
-  const [feeAmount, setFeeAmount] = useState(false);
-  const [donationAmount, setDonationAmount] = useState(false);
-  const [totalAmount, setTotalAmount] = useState(false);
+  const [selectedColumns, setSelectedColumns] = useState<Set<ApplicationsReportColumn>>(new Set());
+
+  // A function that returns an event-handler function that adds/removes the column from selectedColumns
+  const handleSelectColumn =
+    (column: ApplicationsReportColumn) => (event: ChangeEvent<HTMLInputElement>) => {
+      const updatedSelectedColumns = new Set([...selectedColumns]);
+      event.target.checked
+        ? updatedSelectedColumns.add(column)
+        : updatedSelectedColumns.delete(column);
+      setSelectedColumns(updatedSelectedColumns);
+    };
+
+  const handleSelectAllColumns = (event: ChangeEvent<HTMLInputElement>) => {
+    const updatedSelectedColumns = event.target.checked
+      ? new Set([
+          ApplicationsReportColumn.UserId,
+          ApplicationsReportColumn.ApplicantName,
+          ApplicationsReportColumn.ApplicantDateOfBirth,
+          ApplicationsReportColumn.AppNumber,
+          ApplicationsReportColumn.ApplicationDate,
+          ApplicationsReportColumn.PaymentMethod,
+          ApplicationsReportColumn.FeeAmount,
+          ApplicationsReportColumn.DonationAmount,
+          ApplicationsReportColumn.TotalAmount,
+        ])
+      : new Set([]);
+    setSelectedColumns(updatedSelectedColumns);
+  };
 
   // Toast message
   const toast = useToast();
 
   // Export csv query
   const [exportCSV, { loading }] = useLazyQuery<
-    GenerateApplicantsReportResponse,
-    GenerateApplicantsReportRequest
-  >(GENERATE_APPLICANTS_REPORT_QUERY, {
+    GenerateApplicationsReportResponse,
+    GenerateApplicationsReportRequest
+  >(GENERATE_APPLICATIONS_REPORT_QUERY, {
+    onCompleted: data => {
+      if (data?.generateApplicationsReport.ok) {
```

Comment:
```suggestion
      if (data.generateApplicationsReport.ok) {
```
Data will be truthy on complete